File: 25.07.2024/Inventory Management.java
import java.util.HashMap;
import java.util.Map;
class Product {
    private String name;
    private int quantity;
    private double price;
    public Product(String name, int quantity, double price) {
        this.name = name;
        this.quantity = quantity;
        this.price = price;
    }
    @Override
    public String toString() {
        return name + " (Qty: " + quantity + ", Price: $" + price + ")";
    }
    public void updateQuantity(int soldQuantity) {
        if (quantity >= soldQuantity) {
            quantity -= soldQuantity;
            System.out.println("Updated quantity for " + name + ": " + quantity);
        } else {
            System.out.println("Insufficient stock for " + name);
        }
    }
    public static void main(String[] args) {
        Product product1 = new Product("Widget", 100, 10.0);
        Product product2 = new Product("Gadget", 50, 20.0);
        product1.updateQuantity(20);
        product2.updateQuantity(60);
        System.out.println(product1);
        System.out.println(product2);
    }
}

File: 25.07.2024/Inventory Management.c
#include <stdio.h>
#include <string.h>
struct Product {
    char name[50];
    int quantity;
    double price;
};
void updateQuantity(struct Product *product, int soldQuantity) {
    if (product->quantity >= soldQuantity) {
        product->quantity -= soldQuantity;
        printf("Updated quantity for %s: %d\n", product->name, product->quantity);
    } else {
        printf("Insufficient stock for %s\n", product->name);
    }
}
int main() {
    struct Product product1 = {"Widget", 100, 10.0};
    struct Product product2 = {"Gadget", 50, 20.0};
    updateQuantity(&product1, 20);
    updateQuantity(&product2, 60);
    printf("%s (Qty: %d, Price: $%.2lf)\n", product1.name, product1.quantity, product1.price);
    printf("%s (Qty: %d, Price: $%.2lf)\n", product2.name, product2.quantity, product2.price);
    return 0;
}
